mso/resource_mso_tenant_policies_ipsla_monitoring_policy.go

```diff
@@ -0,0 +1,307 @@
+			"http_uri": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"sla_frequency": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validation.IntBetween(1, 300),
+			},
+			"detect_multiplier": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validation.IntBetween(1, 100),
+			},
+			"request_data_size": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validation.IntBetween(1, 17512),
+			},
+			"type_of_service": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validation.IntBetween(0, 255),
+			},
+			"operation_timeout": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validation.IntBetween(0, 604800000),
+			},
+			"threshold": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validation.IntBetween(0, 604800000),
+			},
+			"ipv6_traffic_class": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validation.IntBetween(0, 255),
+			},
+		},
+		CustomizeDiff: func(diff *schema.ResourceDiff, v interface{}) error {
+			slaType, _ := diff.GetOk("sla_type")
+			if slaType == "icmp" || slaType == "l2ping" {
+				diff.SetNew("destination_port", 0)
+			} else if slaType == "http" {
+				diff.SetNew("destination_port", 80)
+			}
+			return nil
+		},
+	}
+}
+
+func setIPSLAMonitoringPolicyData(d *schema.ResourceData, response *container.Container, templateId string) error {
+
+	d.SetId(fmt.Sprintf("templateId/%s/IPSLAMonitoringPolicy/%s", templateId, models.StripQuotes(response.S("name").String())))
+	d.Set("template_id", templateId)
+	d.Set("name", models.StripQuotes(response.S("name").String()))
+	d.Set("description", models.StripQuotes(response.S("description").String()))
+	d.Set("sla_type", models.StripQuotes(response.S("slaType").String()))
+	d.Set("destination_port", response.S("slaPort").Data().(float64))
+	d.Set("http_version", models.StripQuotes(response.S("httpVersion").String()))
+	d.Set("http_uri", models.StripQuotes(response.S("httpUri").String()))
+	d.Set("sla_frequency", response.S("slaFrequency").Data().(float64))
+	d.Set("detect_multiplier", response.S("slaDetectMultiplier").Data().(float64))
+	d.Set("request_data_size", response.S("reqDataSize").Data().(float64))
+	d.Set("type_of_service", response.S("ipv4ToS").Data().(float64))
+	d.Set("operation_timeout", response.S("timeout").Data().(float64))
+	d.Set("threshold", response.S("threshold").Data().(float64))
+	d.Set("ipv6_traffic_class", response.S("ipv6TrfClass").Data().(float64))
+	d.Set("uuid", models.StripQuotes(response.S("uuid").String()))
+
+	return nil
+
+}
+
+func buildMSOIPSLAMonitoringPolicyPayload(d *schema.ResourceData) map[string]interface{} {
```

Comment:
Should we name this consistently create? think this was used in the template resource. Changing the other to build also is ok by me.

Reply:
Removed this